daemon/src/model.rs

```diff
@@ -600,6 +602,73 @@ impl Timestamp {
     }
 }
 
+/// Funding rate per SETTLEMENT_INTERVAL
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct FundingRate(pub Decimal);
+
+impl From<Decimal> for FundingRate {
+    fn from(decimal: Decimal) -> Self {
+        FundingRate(decimal)
+    }
+}
+
+/// Funding parameters used for calculating the fees
+#[derive(Debug, Clone)]
+pub struct FundingParams {
+    funding_rate: FundingRate,
+    /// How much of the funding round are we aiming to pay for now
+    fraction: Decimal,
+    amount_sat: Decimal,
+}
+
+impl Default for FundingParams {
+    fn default() -> Self {
+        Self {
+            amount_sat: Decimal::ZERO, // Amount doesn't matter - fee will be 0 anyway
+            funding_rate: FundingRate(Decimal::ZERO),
+            fraction: Decimal::ONE,
+        }
+    }
+}
+
+impl FundingParams {
+    pub fn new(amount: Amount, funding_rate: FundingRate, hours_to_charge: i64) -> Result<Self> {
+        anyhow::ensure!(
+            funding_rate.0.is_sign_positive(),
+            "Negative funding rates (ie. paid by the maker) are not supported yet"
+        );
```

Comment:
If `FundingRate` were to make its field private, you could enforce this in the constructor and rely on it here!